packages/adblocker-extended-selectors/src/eval.ts

```diff
@@ -71,104 +71,247 @@ export function matches(element: Element, selector: AST): boolean {
       }
 
       return text.length >= minLength;
+    } else if (selector.name === 'matches-path') {
+      const { argument } = selector;
+      if (argument === undefined) {
+        return false;
+      }
+
+      const path = globalThis.window.location.pathname;
```

Comment:
The documentation from uBO specifies that this should match on path + query but here we only consider path. See: https://github.com/gorhill/uBlock/wiki/Procedural-cosmetic-filters#subjectmatches-patharg